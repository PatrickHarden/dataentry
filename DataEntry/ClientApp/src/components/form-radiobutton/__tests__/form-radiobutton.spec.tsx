import React from 'react';
import { mount, shallow } from 'enzyme';

import FormRadioButton from '../form-radiobutton';

describe('components', () => {

  describe('form-radiobutton', () => {

    let props:any;
    let wrapper:any;

    beforeEach(() => {
        props = {
            name: 'testme',
            label: 'Turn up the radio!',
            defaultValue: '',
            error: false,
            changeHandler: jest.fn()
        }
    });

    // has proper label
    it("shows its label properly", () => {
        wrapper = mount(<FormRadioButton {...props}/>);
        expect(wrapper.find('h5').text()).toEqual("Turn up the radio!");
    })

    // snapshot render
    it('renders according to snapshot', () => {
        expect(shallow(<FormRadioButton {...props} />)).toMatchSnapshot();
    });
  });
});