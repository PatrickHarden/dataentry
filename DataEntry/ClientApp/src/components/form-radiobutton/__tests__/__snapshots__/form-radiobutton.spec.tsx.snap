// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components form-radiobutton renders according to snapshot 1`] = `
<styled.div
  error={false}
  onMouseDown={[Function]}
>
  <styled.label>
    <styled.input
      checked={false}
      className="radio-btn"
      id="testme_undefined"
      name="testme"
      readOnly={true}
      type="radio"
    />
    <FormFieldLabel
      error={false}
      label="Turn up the radio!"
    />
  </styled.label>
</styled.div>
`;
